spec/services/hashtag_autocomplete_service_spec.rb

```diff
@@ -166,6 +170,19 @@ def self.search(guardian_scoped, term, limit)
       )
     end
 
+    it "orders categories by exact match (ignoring parent/child distinction) and name then name for everything else" do
+      category2 = Fabricate(:category, name: "Book Library", slug: "book-library")
+      category3 = Fabricate(:category, name: "Horror", slug: "book", parent_category: category2)
+      category4 = Fabricate(:category, name: "Romance", slug: "romance-books")
+      category5 =
+        Fabricate(:category, name: "Abstract Philosophy", slug: "abstract-philosophy-books")
+      category6 = Fabricate(:category, name: "Book Reviews", slug: "book-reviews")
+      category7 = Fabricate(:category, name: "Good Books", slug: "book", parent_category: category6)
+      expect(subject.search("book", %w[category]).map(&:ref)).to eq(
+        %w[book-reviews:book book-library:book abstract-philosophy-books book-club book-library book-reviews romance-books],
```

Comment:
Something that isn't clear to is why `book-reviews:book` is sorted above `book-library:book`. If I sort by the name of the child category or the parent category, `book-library:book` should come before `book-reviews:book`.

Reply:
Yes it's tricky... `book-reviews:book` is the ref for the category with the name `Good Books` and `book-library:book` is the ref for the category with the name `Horror`, so that is the reason for the sorting (G before H). It's sorting by the name of the category itself. So we do:

1. Sort by exact slug match regardless of parent/child, then within that sort by category name
2. Sort by remaining non-exact match category names

Reply:
ooo icic thank you for clarifying :+1:

Reply:
No problem, I was even confusing myself with this when writing the spec yesterday 😆 I will actually add a second line to this spec that shows the `:text:` version as well so it is clearer